src/student_success_tool/analysis/pdp/targets/failure_to_graduate_within_x_years.py

```diff
@@ -0,0 +1,142 @@
+import logging
+import typing as t
+from collections.abc import Collection
+import pandas as pd
+from student_success_tool.analysis.pdp.targets import shared
+from student_success_tool.analysis.pdp import utils
+
+LOGGER = logging.getLogger(__name__)
+
+
+def make_labeled_dataset(
+    df: pd.DataFrame,
+    *,
+    student_criteria: dict[str, object | Collection[object]],
+    student_id_cols: str | list[str] = "student_guid",
+    exclude_pre_cohort_terms: bool = True,
+    term_is_pre_cohort_col: str = "term_is_pre_cohort",
+    years_to_bach_at_cohort_col: str = "first_year_to_bachelors_at_cohort_inst",
+    max_years_to_graduation: int = 4,
+    intensity_time_lefts: list[tuple[str, float, t.Literal["year", "term"]]],
```

Comment:
re @bdewilde quoting you: 
"These arguments both represent the same thing, no? I'd recommend getting rid of max_years_to_graduation and using the more flexible data structure so you can specify max years differently between part- and full-time students." - yes this is a great idea and I will make that change soon!